### PR TITLE
Scrolling student list

### LINKED ISSUE #76: Add scrolling to student list in Create Assignment page
Right now the drop down component we have looks fine, but once there's like 10+ students, it'll look strange, so we should add scrolling to the component or use another selector
image

### PR DESCRIPTION
**Pull request by AngelaLuo49021**
## Status:

:rocket: Ready

## Description :star2:

Fixes #76 
Lowkey doesn't really look like Anisha's design but I personally think its better because 3 columns for really tiny usernames doesn't really make sense. (or mayb im just saying that bc CSS is hard). Also theres like an info button next to every single student and i have no idea what thats about so i just decided to not include it. Please merge this in before she sees this lol. Lowkey if i ever become a product designer, id make everything hella easy to code bc CSS is suffering but then maybe everything i design will be ugly???

Also check these comments on line 66 of Student Selector
    //So for optimization, the select button works differently here than it does for the word group select all
    //if we want to make it be the same, for each render, it will need to check if all the students in that deployment has been selected

Basically in word groups right now, the select button is checked when all the words are selected (even if done manually) This is kept track of by just checking the number of words selected and if thats equal to the total number of words in that group. 

However, in student selector, can't use the above method because the select all button is unique to the current class and the list of students will include students from many different classes. So for each render, it can check if all the students in the current class are in the checked list (gross) or if someone can think of something better thatd be good. Tbh i dont think its a big a deal since its still intuitive and if anything, we can change the word group select all button to match this one. 

## TODOs :star:

Screen Shot 2020-04-29 at 1 56 53 AM
Screen Shot 2020-04-29 at 1 57 34 AM
Screen Shot 2020-04-29 at 1 57 39 AM

### CODE REVIEW

File: src/components/StudentSelector/StudentSelector.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    const setChecked = Array.from(new Set(newChecked));
+    setStudentsChecked(setChecked);
+    props.handleChange(setChecked);
+    console.log(setChecked);
+
+    let newChooseAll = [...chooseAll];
+    newChooseAll[index] = !newChooseAll[index];
+    setChooseAll(newChooseAll);
+  }
+  //the key in the DropdownButton is hacky fix to the issue that the dropdown menu doesnt close upon selection
```
Comment (chyku) [inline src/components/StudentSelector/StudentSelector.js]: can you not make the key the current deployment's ID or does it specifically have to be the date time

File: src/components/StudentSelector/StudentSelector.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  const [currentDeployment, setCurrentDeployment] = useState();
+  const [studentsChecked, setStudentsChecked] = useState([]);
+  const [chooseAll, setChooseAll] = useState(
+    Array(props.deployments.length)
+      .fill()
+      .map((_, i) => false)
+  );
+  let deployments = props.deployments;
+
+  function nameSelector(deploymentAccountId) {
```
Comment (chyku) [inline src/components/StudentSelector/StudentSelector.js]: can u name this NameSelector & follow that convention for any other functions that are basically components

File: src/components/StudentSelector/StudentSelector.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+function StudentSelector(props) {
+  const [currentDeployment, setCurrentDeployment] = useState();
+  const [studentsChecked, setStudentsChecked] = useState([]);
+  const [chooseAll, setChooseAll] = useState(
+    Array(props.deployments.length)
+      .fill()
+      .map((_, i) => false)
+  );
+  let deployments = props.deployments;
```
Comment (chyku) [inline src/components/StudentSelector/StudentSelector.js:15]: oh so this is problematic if this depends on a promise ic why you cant set a default deployment

Comment (chyku) [reply src/components/StudentSelector/StudentSelector.js:15]: If possible if no class is chosen it should say `Select a Class` to indicate that there's no class

File: src/components/StudentSelector/StudentSelector.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+  function nameSelector(deploymentAccountId) {
+    return (
+      <div className="select-all-margins">
+        <Row>
+          <div>
+            <input
+              class="move-down"
```
Comment (chyku) [inline src/components/StudentSelector/StudentSelector.js]: what does move-down mean as a class name

File: src/components/StudentSelector/StudentSelector.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        newChecked.splice(removeIndex, 1);
+      }
+    });
+    //So for optimization, the select button works differently here than it does for the word group select all
+    //if we want to make it be the same, for each render, it will need to check if all the students in that deployment has been selected
+
+    const setChecked = Array.from(new Set(newChecked));
+    setStudentsChecked(setChecked);
+    props.handleChange(setChecked);
+    console.log(setChecked);
```
Comment (chyku) [inline src/components/StudentSelector/StudentSelector.js]: pls no console logs

File: src/components/StudentSelector/StudentSelector.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+    if (currentIndex === -1) {
+      newChecked.push(deploymentAccountId);
+    } else {
+      newChecked.splice(currentIndex, 1);
+    }
+    const setChecked = Array.from(new Set(newChecked));
+    setStudentsChecked(setChecked);
+    props.handleChange(setChecked);
+    console.log(setChecked);
```
Comment (chyku) [inline src/components/StudentSelector/StudentSelector.js]: pls no console.logs

File: src/components/StudentSelector/StudentSelector.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+  }
+
+  function handleClickSelectAll(deployment) {
+    let index = deployments.indexOf(currentDeployment);
+    let deploymentAccounts = [];
+    let selectAll = false;
+    let newChecked = [...studentsChecked];
+    let indexes = [];
```
Comment (chyku) [inline src/components/StudentSelector/StudentSelector.js]: i think some of these variables aren't used in this function so pls delete them

**Review by chyku — CHANGES_REQUESTED** [review]
sum things

File: src/components/StudentSelector/StudentSelector.js (hunk 4 above)
Comment (AngelaLuo49021) [reply src/components/StudentSelector/StudentSelector.js]: lol thats not even supposed to be there

**Review by chyku — APPROVED** [review]

**Pull request merged**